src/Workspaces/Core/Portable/Execution/PinnedRemotableDataScope.cs

```diff
@@ -10,15 +10,46 @@
 namespace Microsoft.CodeAnalysis.Execution
 {
     /// <summary>
+    /// Information related to pinned solution
+    /// </summary>
+    internal class PinnedSolutionInfo
+    {
+        /// <summary>
+        /// Unique ID for this pinned solution
+        /// 
+        /// This later used to find matching solution between VS and remote host
+        /// </summary>
+        public readonly int ScopeId;
+
+        /// <summary>
+        /// This indicates whether this scope is for primary branch or not (not forked solution)
+        /// 
+        /// Features like OOP will use this flag to see whether caching information related to this solution
+        /// can benefit other requests or not
+        /// </summary>
```

Comment:
i would thinking caching would be generally valuable regardless of what solution it was. for example, if a feature forks a solution, then still, the vast majority of what it would cache would still likely be vlauable across other requests.  

I'm thinking about something like Rename which will fork things a few times.  It's likely we'd like to cache intermediary values so that later requests don't have to resync the same things over.